app/services/github_service.rb

```diff
@@ -13,41 +13,37 @@ def call
   private
 
   def handle_request
-    build_project
     handle_event
-    raise Events::NotHandleableError unless @entity
-
     handle_action
   end
 
-  def build_project
-    repo = @payload['repository']
-    @project = Project.find_or_create_by!(github_id: repo['id']) do |pj|
-      pj.name = repo['name']
-      pj.description = repo['description']
-    end
-  end
-
   def handle_event
-    @entity = find_or_create_event_type
-
-    Event.create!(project: @project,
-                  data: @payload,
-                  name: @event,
-                  handleable: @entity,
-                  occurred_at: occurence_time(event_name: @event, payload: @payload))
+    project = Projects::Builder.fetch_or_create(@payload['repository'])
+    event = Event.new(project: project, data: @payload, name: @event)
+    if handleable_event?
+      @entity = find_or_create_event_type
+      event.tap do |evnt|
```

Comment:
I think you don't need to use `tap` here

Reply:
I need to return the `event` here, and `.save` does not return the object, instead it returns `true` or `false`.

Reply:
you still have event, this would be the same
```
  event.assign_attributes(handleable: @entity, occurred_at: occurence_time)
  event.save!
  event
```

Reply:
Alright